CMakeLists.txt

```diff
@@ -27,8 +27,23 @@ enable_testing()
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-# torch, for which we are going to rely on the system for now
-find_package(Torch REQUIRED)
+# Location of current CPU torch
+include(FetchContent)
+set(FETCHCONTENT_BASE_DIR "${CMAKE_SOURCE_DIR}/extern")
+FetchContent_Declare(
+  torch
+  URL https://download.pytorch.org/libtorch/nightly/cpu/libtorch-shared-with-deps-latest.zip
+)
+
+# Select between system torch and the current (CPU) torch
+option(SYSTEM_TORCH "Whether to use the system torch or download the current CPU version" OFF)
+if (SYSTEM_TORCH)
+      set(TORCH_DIR "" CACHE PATH "Path to a system installation of torch")
+      find_package(Torch REQUIRED HINTS "${TORCH_DIR}/share/cmake")
+else()
+      FetchContent_MakeAvailable(torch)
+      find_package(Torch REQUIRED HINTS "${torch_SOURCE_DIR}/share/cmake")
+endif()
```

Comment:
I recommend consolidating `SYSTEM_TORCH` and `TORCH_DIR` into one variable, and just check `if(NOT DEFINED TORCH_DIR)`.

Reply:
Also rename `TORCH_DIR` to `LIBTORCH_DIR` to prevent confusion, i.e., should I point it to pytorch or libtorch? Same with the variable description, we can be more precise by saying "Path to a custom libtorch's root directory such that LIBTORCH_DIR/share/cmake exists".